src/runtime/cli/kata-exec.go

```diff
@@ -168,15 +170,20 @@ func (s *iostream) Read(data []byte) (n int, err error) {
 	return s.conn.Read(data)
 }
 
-func getConn(endPoint, sandboxID string, port uint64) (net.Conn, error) {
-	shimURL := fmt.Sprintf("%s/agent-url?sandbox=%s", endPoint, sandboxID)
-	resp, err := http.Get(shimURL)
+func getConn(namespace, sandboxID string, port uint64) (net.Conn, error) {
+	socketAddr := fmt.Sprintf("/containerd-shim/%s/%s/shim-monitor.sock", namespace, sandboxID)
```

Comment:
Does the path follow a common pattern? I wonder why there is a namespace `k8s` in the middle

Reply:
Yes, the path format is fixed: https://github.com/kata-containers/kata-containers/blob/main/src/runtime/containerd-shim-v2/shim_management.go#L196

And the namespace is passed from containerd: https://github.com/kata-containers/kata-containers/blob/main/src/runtime/vendor/github.com/containerd/containerd/runtime/v2/shim/shim.go#L89 , the default value  is k8s.io

Reply:
How does this affect setups not using containerd or k8s?  e.g. a standalone CRI-O setup?

Reply:
@dgibson the address is independent of runtimes, it only depends on the namespace passed from the runtime.

Reply:
It's safer to use `filepath.Join()` / `path.Join()` here.